bundle/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/Flamingos2Epics.scala

```diff
@@ -99,26 +100,27 @@ object Flamingos2Epics {
 
 }
 
-object Flamingos2EpicsInitializer {
+object Flamingos2Epics extends EpicsSystem {
+  
   val Log = Logger.getLogger(getClass.getName)
   val CA_CONFIG_FILE = "/Flamingos2.xml"
+  
+  // Ugly use of null. But if it is used before initialization, then the application deserves to crash.
+  var instance: Flamingos2Epics = null
 
-  def init: TrySeq[Unit] = {
+  override def init(service: CaService): TrySeq[Unit] = {
     try {
       (new XMLBuilder).fromStream(this.getClass.getResourceAsStream(CA_CONFIG_FILE))
```

Comment:
No, it creates a series of command senders and status acceptors in CaService, which then Flamingos2Epics retrieves by name.

The XML configuration files are created from the old CA files in Seqexec with a little tool I wrote. They have the list of all EPICS channels that are read or written by the application. I certain improvement would be to have the tool generate code to create all the command senders and status acceptors, allowing for direct references and compile checks. Hopefully in the (near) future...